EPPs/common.py

```diff
@@ -144,6 +153,86 @@ def run(self):
             self.error('Stack trace below:\n' + stacktrace)
             raise e
 
+    @cached_property
+    def input_container_names(self):
+        """The name of containers from input artifacts. """
+        containers = set()
+        for art in self.artifacts:
+            # Check to see if artifact has a container before retrieving the container.
+            # Artifacts that are not samples will not have containers.
+            if art.container:
+                containers.add(art.container.name)
+        return list(containers)
```

Comment:
Might we end up with non-deterministic order here?

Reply:
Should be deterministic now